src/main/scala/cromwell/engine/backend/local/LocalBackend.scala

```diff
@@ -135,16 +133,19 @@ class LocalBackend extends Backend with LazyLogging {
    */
   private def stageWorkflowInputs(descriptor: WorkflowDescriptor): HostInputs = {
     val hostInputsPath = Paths.get(hostExecutionPath(descriptor).toFile.getAbsolutePath, "workflow-inputs")
-    descriptor.actualInputs map {stageInput(_, hostInputsPath)}
+    descriptor.actualInputs.map {case(name, value) => stageInput(name, value, hostInputsPath)}
   }
 
-  private def stageInput(nameAndValue: (String, WdlValue), hostInputsPath: Path): (String, WdlValue) = {
-    val (name, value) = nameAndValue
-    val hostPathAdjustedValue = value match {
-      case w: WdlFile => stageWdlFile(w, hostInputsPath)
-      case x => x
-    }
-    name -> hostPathAdjustedValue
+  private def stageInputArray(array: WdlArray, hostInputsPath: Path): WdlArray = array.map(stageWdlValue(_, hostInputsPath))
+
+  private def stageWdlValue(value: WdlValue, hostInputsPath: Path) = value match {
+    case w:WdlFile => stageWdlFile(w, hostInputsPath)
+    case a:WdlArray => stageInputArray(a, hostInputsPath)
+    case x => x
+  }
+
+  private def stageInput(name: String, value: WdlValue, hostInputsPath: Path): (String, WdlValue) = {
```

Comment:
I'm pretty sure we have a lot of `(String, WdlValue)` tuples floating around. It'd be nice to codify that w/ a case class instead of tupling it up everywhere